apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.command.ts

```diff
@@ -1,10 +1,23 @@
 import { EnvironmentWithUserCommand } from '@novu/application-generic';
 import { ActionStepEnum, ChannelStepEnum, StepType } from '@novu/framework';
-import { IsEnum } from 'class-validator';
+import { IsEnum, IsNotEmpty, IsString } from 'class-validator';
 
 const StepTypeValue = { ...ChannelStepEnum, ...ActionStepEnum };
 
-export class GetStepSchemaCommand extends EnvironmentWithUserCommand {
+export class GetStepTypeSchemaCommand extends EnvironmentWithUserCommand {
   @IsEnum(StepTypeValue)
+  @IsNotEmpty()
   stepType: StepType;
 }
+
+export class GetExistingStepSchemaCommand extends EnvironmentWithUserCommand {
+  @IsString()
+  @IsNotEmpty()
+  workflowId: string;
+
+  @IsString()
+  @IsNotEmpty()
+  stepId: string;
+}
+
+export type GetStepSchemaCommand = GetStepTypeSchemaCommand | GetExistingStepSchemaCommand;
```

Comment:
Could we simplify this command to support multiple use cases and have the internal use case handle the matching and business logic?

Reply:
yes but i think it will be overkill for the logic we have today